tests/app/dao/notification_dao/test_notification_dao.py

```diff
@@ -2034,39 +2039,42 @@ def test_update_notification_status_by_id_cannot_update_status_out_of_order_with
     ],
 )
 def test_update_notification_status_by_id_can_update_status_in_order_when_given_valid_values(
-    current_status,
-    next_status,
     sample_notification,
     sample_template,
+    current_status,
+    next_status,
+    use_current_status,
 ):
-    reference_tuple = (str(uuid4()), str(uuid4()))
-    notification_list = []
-
-    for ref in reference_tuple:
-        # create first notification object
-        sample_notification(template=sample_template(), reference=ref, sent_at=datetime.now(), status=current_status)
-
-        # add the notification object to the notification list
-        notification_list.append(dao_get_notification_by_reference(ref))
-
-        # make sure we pass all the init setup
-        i = len(notification_list) - 1
-        assert isinstance(notification_list[i], Notification)
-        assert notification_list[i].status == current_status
-
-    # attempt update without the condition current state
-    update_notification_status_by_id(notification_id=notification_list[0].id, status=next_status)
+    reference = str(uuid4())
+    initial_status_reason = '' if (current_status == NOTIFICATION_DELIVERED) else 'Because I said so!'
+    final_status_reason = initial_status_reason if (next_status == current_status) else 'just because'
```

Comment:
Please remove this. It always evaluates to false.

Reply:
done